geopandas/array.py

```diff
@@ -967,34 +967,54 @@ def take(self, indices, allow_fill=False, fill_value=None):
         return GeometryArray(result, crs=self.crs)
 
     def _fill(self, idx, value):
-        """Fill index locations with value
+        """
+        Fill index locations with ``value``.
 
-        Value should be a BaseGeometry
+        ``value`` should be a BaseGeometry or a GeometryArray.
         """
-        if not (_is_scalar_geometry(value) or value is None):
+        from pandas import Series
+
+        if vectorized.isna(value):
+            value = [BaseGeometry()]
+        elif _is_scalar_geometry(value):
+            value = [value]
+        elif isinstance(value, GeometryArray):
```

Comment:
I thought `value` could receive `GeometryArray` or `GeoSeries`.
If it's array-like, It would be hard to check whether its element is geometry type or not.